leshan-server-cf/src/main/java/org/eclipse/leshan/server/californium/impl/CaliforniumObservationRegistryImpl.java

```diff
@@ -268,7 +268,7 @@ public void onNotification(Request coapRequest, Response coapResponse) {
                 LwM2mModel model = modelProvider.getObjectModel(client);
 
                 // get content format
-                ContentFormat contentFormat = null;
+                ContentFormat contentFormat = ContentFormat.TEXT;
```

Comment:
I do not think that this is the correct point in the code to fix the problem. At this point we are just dealing with a symptom. The lack of the content format option needs to be detected deeper down the stack when we receive a response from a client. According to the latest spec the content format is mandatory for clients to set in their responses, so we should think about a general strategy for dealing with this. I can imagine to reject or ignore such response messages (the spec doesn't seem to give any hint on this) or trying to *determine* or *guess* the content format from the payload ...